examples/trivia-game-example/src/machines/triviaMachine.ts

```diff
@@ -0,0 +1,245 @@
+import { assign, fromPromise, not, setup } from "xstate";
+import { RMCharacter } from "../common/types";
+import { RickCharacters } from "../services/RickApi";
+import { getRandomNumber } from "../common/constants";;
+
+const triviaMachine = setup({
+    types: {
+        events: {} as 
+        | { type: "user.play"} 
+        | { type: "user.close"} 
+        | { type: "user.reject"}
+        | { type: "user.accept"}
+        | { 
+            type: "user.selectAnswer",
+            answer: number
+        }
+        | { type: "user.nextQuestion"}
+        | { type: "user.toggleClue"}
+        | { type: "user.playAgain"},
+        context: {} as {
+            homePageCharacters: Array<RMCharacter>
+            hasLoaded: boolean
+            currentCharacter: RMCharacter | null
+            randomCharacters: Array<RMCharacter>
+            isClueOpened: boolean
+            points: number
+            question: number
+            lifes: number
+        }
+    },
+    guards: {
+        isAnswerCorrect: ({context, event}) => {
+            if(event.type !== "user.selectAnswer") return false
```

Comment:
Nit: better to use `params` as a future refactor item, but for now, since we're only expecting the guard to be called in one place:

```suggestion
            assertEvent(event, 'user.selectAnswer');
```